File: backend/Hirenix.Domain/Entities/Job.cs
using Hirenix.Domain.Enums;

namespace Hirenix.Domain.Entities;

public class Job
{
    public ulong Id { get; set; }
    public ulong CompanyId { get; set; }
    public ulong CreatedBy { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? Requirements { get; set; }
    public string? Benefits { get; set; }
    public WorkType WorkType { get; set; }
    public JobLevel Level { get; set; }
    public long? SalaryMin { get; set; }
    public long? SalaryMax { get; set; }
    public bool IsSalaryVisible { get; set; } = true;
    public uint? CityId { get; set; }
    public uint? IndustryId { get; set; }
    public DateOnly Deadline { get; set; }
    public JobStatus Status { get; set; } = JobStatus.Draft;
    public bool IsFeatured { get; set; }
    public uint ViewsCount { get; set; }
    public uint ApplicationsCount { get; set; }
    public ulong? ParentJobId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Company? Company { get; set; }
    public EmployerProfile? Creator { get; set; }
    public Location? City { get; set; }
    public Industry? Industry { get; set; }
    public Job? ParentJob { get; set; }
    public ICollection<JobSkill> Skills { get; set; } = new List<JobSkill>();
}

public class JobSkill
{
    public ulong Id { get; set; }
    public ulong JobId { get; set; }
    public uint SkillId { get; set; }
    public bool IsRequired { get; set; } = true;

    // Navigation
    public Job? Job { get; set; }
    public Skill? Skill { get; set; }
}

File: mobile/src/types/job.ts
export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

export interface PaginationMetadata {
  currentPage: number;
  pageSize: number;
  totalPages: number;
  totalItems: number;
}

export interface PaginatedResult<T> {
  data: T[];
  pagination: PaginationMetadata;
}

export interface JobListItemDto {
  id: number;
  title: string;
  companyName: string;
  companyLogo?: string;
  cityName?: string;
  workType: string;
  level: string;
  salaryMin?: number;
  salaryMax?: number;
  isSalaryVisible: boolean;
  deadline: string;
  status: string;
  isFeatured: boolean;
  viewsCount: number;
  applicationsCount: number;
  skills: string[];
  createdAt: string;
}

export interface FilterLocationOptionDto {
  id: number;
  name: string;
}

export interface FilterIndustryOptionDto {
  id: number;
  name: string;
}

export interface FilterWorkTypeOptionDto {
  value: number;
  name: string;
}

export interface FilterLevelOptionDto {
  value: number;
  name: string;
}

export interface FilterOptionsDto {
  cities: FilterLocationOptionDto[];
  industries: FilterIndustryOptionDto[];
  workTypes: FilterWorkTypeOptionDto[];
  levels: FilterLevelOptionDto[];
}

export interface JobDetailCompanyDto {
  id: number;
  name: string;
  logoUrl?: string;
  website?: string;
  description?: string;
}

export interface JobDetailSkillDto {
  id: number;
  name: string;
  isRequired: boolean;
}

export interface JobDetailDto {
  id: number;
  title: string;
  description: string;
  requirements?: string;
  benefits?: string;
  workType: string;
  level: string;
  salaryMin?: number;
  salaryMax?: number;
  isSalaryVisible: boolean;
  deadline: string;
  status: string;
  isFeatured: boolean;
  viewsCount: number;
  applicationsCount: number;
  createdAt: string;
  updatedAt: string;
  company: JobDetailCompanyDto;
  city?: string;
  industry?: string;
  skills: JobDetailSkillDto[];
  hasApplied: boolean;
}

export interface JobFilterParams {
  search?: string;
  cityId?: number;
  industryId?: number;
  sortBy?: 'CreatedAt' | 'salary' | 'title' | 'deadline';
  page?: number;
  pageSize?: number;
}

export interface SubmitApplicationDto {
  jobId: number;
  cvFileUri: string;
  cvFileName: string;
  coverLetter?: string;
}
